packages/common/src/contexts/wallet.tsx

```diff
@@ -1,241 +1,160 @@
-  const close = useCallback(() => setIsModalVisible(false), []);
+    setConnected(!!publicKey);
+  }, [publicKey, connected, setConnected]);
 
   return (
-    <WalletContext.Provider
+    <WalletModalContext.Provider
       value={{
-        wallet,
-        connected,
-        select,
-        provider,
+        visible,
+        setVisible,
       }}
     >
       {children}
-      <Modal
-        title="Select Wallet"
-        okText="Connect"
-        visible={isModalVisible}
-        okButtonProps={{ style: { display: 'none' } }}
-        onCancel={close}
-        width={400}
-      >
-        {WALLET_PROVIDERS.map((provider, idx) => {
-          const onClick = function () {
-            setProviderUrl(provider.url);
-            setAutoConnect(true);
-            close();
-          };
+      <WalletModal/>
+    </WalletModalContext.Provider>
+  );
+};
 
-          return (
-            <Button
-              key={idx}
-              size="large"
-              type={providerUrl === provider.url ? 'primary' : 'ghost'}
-              onClick={onClick}
-              icon={
-                <img
-                  alt={`${provider.name}`}
-                  width={20}
-                  height={20}
-                  src={provider.icon}
-                  style={{ marginRight: 8 }}
-                />
-              }
-              style={{
-                display: 'block',
-                width: '100%',
-                textAlign: 'left',
-                marginBottom: 8,
-              }}
-            >
-              {provider.name}
-            </Button>
-          );
-        })}
-      </Modal>
-    </WalletContext.Provider>
+export const WalletProvider: FC<{children: ReactNode }> = ({ children }) => {
+  const wallets = useMemo(
+    () => [
+      getPhantomWallet(),
+      getSolflareWallet(),
+      getTorusWallet({
+        options: {
+          clientId: 'BOM5Cl7PXgE9Ylq1Z1tqzhpydY0RVr8k90QQ85N7AKI5QGSrr9iDC-3rvmy0K_hF0JfpLMiXoDhta68JwcxS1LQ',
+        },
+      }),
+      getLedgerWallet(),
+      getSolongWallet(),
+      getMathWallet(),
+      getSolletWallet(),
+    ],
+    []
+  );
+
+  const onError = useCallback((error: WalletError) => {
+    console.error(error);
+    notify({
+      message: 'Wallet error',
+      description: error.message,
+    });
+  }, []);
+
+  return (
+    <BaseWalletProvider
+      wallets={wallets}
+      onError={onError}
+      autoConnect
+    >
+      <WalletModalProvider>
+        {children}
+      </WalletModalProvider>
+    </BaseWalletProvider>
   );
 }
 
-export const useWallet = () => {
-  const { wallet, connected, provider, select } = useContext(WalletContext);
-  return {
-    wallet,
-    connected,
-    provider,
-    select,
-    connect() {
-      wallet ? wallet.connect() : select();
-    },
-    disconnect() {
-      wallet?.disconnect();
-    },
-  };
-};
+export type WalletSigner = Pick<WalletAdapter, 'publicKey' | 'signTransaction' | 'signAllTransactions'>;
```

Comment:
The minimal interface of the context and adapter needed to sign transactions.

Reply:
+1 like